runtime/startup_byt.c

```diff
@@ -273,56 +274,176 @@ static char_os * read_section_to_os(int fd, struct exec_trailer *trail,
+          i++;
+        } else {
+          error("option '-I' needs an argument.");
+        }
+        break;
+      case 'm':
+        print_magic = 1;
+        break;
+      case 'M':
+        printf("%s\n", EXEC_MAGIC);
+        exit(0);
+        break;
+      default:
+        parsed = 0;
       }
-      break;
-    case 'p':
-      for (j = 0; caml_names_of_builtin_cprim[j] != NULL; j++)
-        printf("%s\n", caml_names_of_builtin_cprim[j]);
-      exit(0);
-      break;
-    case 'b':
-      caml_record_backtrace(Val_true);
-      break;
-    case 'I':
-      if (argv[i + 1] != NULL) {
-        caml_ext_table_add(&caml_shared_libs_path, argv[i + 1]);
-        i++;
+    } else {
+      if (!strcmp_os(argv[i], T("-version"))) {
+        printf("%s\n", "The OCaml runtime, version " OCAML_VERSION_STRING);
+        exit(0);
+      } else if (!strcmp_os(argv[i], T("-vnum"))) {
+        printf("%s\n", OCAML_VERSION_STRING);
+        exit(0);
+      } else if (!strcmp_os(argv[i], T("-help")) ||
+                 !strcmp_os(argv[i], T("--help"))) {
+        do_print_help();
+        exit(0);
+      } else if (!strcmp_os(argv[i], T("-config"))) {
+        print_config = 1;
+      } else if (argv[i][1] == 'I') {
+        caml_ext_table_add(&caml_shared_libs_path, &argv[i][2]);
+      } else {
+        parsed = 0;
       }
-      break;
-    case 'm':
-      print_magic = 1;
-      break;
-    case 'M':
-      printf ( "%s\n", EXEC_MAGIC);
-      exit(0);
-      break;
-    default:
-      error("unknown option %s", caml_stat_strdup_of_os(argv[i]));
     }
+
+    if (!parsed)
+      error("unknown option %s", caml_stat_strdup_of_os(argv[i]));
   }
+
   return i;
 }
 
+/* Print the configuration of the runtime to stdout; memory allocated is not
+   freed, since the runtime will terminate after calling this. */
+static void do_print_config(void)
+{
+  int i;
+  char_os * dir;
+
+  /* Print the runtime configuration */
+  printf("version: %s\n"
+         "standard_library_default: %s\n"
+         "standard_library: %s\n"
+         "int_size: %d\n"
+         "word_size: %d\n"
+         "os_type: %s\n"
+         "host: %s\n"
+         "flat_float_array: %s\n"
+         "supports_afl: %s\n"
+         "windows_unicode: %s\n"
+         "supports_shared_libraries: %s\n"
+         "no_naked_pointers: %s\n"
+         "profinfo: %s\n"
+         "profinfo_width: %d\n"
+         "exec_magic_number: %s\n",
+           OCAML_VERSION_STRING,
+           caml_stat_strdup_of_os(OCAML_STDLIB_DIR),
+           caml_stat_strdup_of_os(caml_get_stdlib_location()),
+           8 * (int)sizeof(value),
+           8 * (int)sizeof(value) - 1,
+           OCAML_OS_TYPE,
+           HOST,
+#ifdef FLAT_FLOAT_ARRAY
+           "true",
+#else
+           "false",
+#endif
+#if HAS_SYS_SHM_H
+           "true",
+#else
+           "false",
+#endif
+#if WINDOWS_UNICODE
+           "true",
+#else
+           "false",
+#endif
```

Comment:
To be really explicit:
```suggestion
#if !defined(WINDOWS_UNICODE)
           "n/a",
#elif WINDOWS_UNICODE
           "true",
#else
           "false",
#endif
```

Reply:
This would be inconsistent with `ocamlc -config` though which has always displayed `false` on Unix